File: check2.py
import sys
sys.stdout.reconfigure(encoding='utf-8')
PATH = 'C:/Users/Admin/.openclaw/workspace-skills/web-builder/sites/kiem-tra-gia/index.html'
with open(PATH, 'r', encoding='utf-8') as f:
    text = f.read()

# Check: is header inside JS or static HTML?
# Find the main script tags
last_script_open = text.rfind('<script>' + '\n' if text[text.rfind('<script>'):text.rfind('<script>')+9] == '<script>\n' else '')
# Find all script tags
import re
for m in re.finditer(r'<script[^>]*>', text):
    pos = m.start()
    # Find matching </script>
    end = text.find('</script>', pos)
    snippet = text[m.start():end]
    print(f'<script> at {m.start()}-{end}: starts with {snippet[:50]}...')
    print(f'  contains "render()": {"render()" in snippet}')
    
# Check first 30 lines
lines = text.split('\n')
print('\n=== First 30 lines ===')
for i in range(min(30, len(lines))):
    print(f'{i+1}: {repr(lines[i][:100])}')

File: _check2.js
const fs = require('fs');
let h = fs.readFileSync('sites/kiem-tra-gia/index.html', 'utf8');

// Check what the actual strings look like in updateQuotationPreview
var idx = h.indexOf('function updateQuotationPreview()');
var end = h.indexOf('// ====== COPY QUOTATION', idx);
var content = h.substring(idx, end + 1);

// Show each Vietnamese string found
var lines = content.split('\n');
for (let i = 0; i < lines.length; i++) {
  var line = lines[i];
  // Check for Vietnamese text patterns
  if (
    line.indexOf('BÁO GIÁ') > -1 ||
    line.indexOf('Khách hàng') > -1 ||
    line.indexOf('Người liên hệ') > -1 ||
    line.indexOf('Người phụ trách') > -1 ||
    line.indexOf('Ngày') > -1 ||
    line.indexOf('Cảng đi') > -1 ||
    line.indexOf('Điều kiện') > -1 ||
    line.indexOf('Hiệu lực') > -1 ||
    line.indexOf('Thanh toán') > -1 ||
    line.indexOf('Ghi chú') > -1 ||
    line.indexOf('Tấn') > -1 ||
    line.indexOf('Tên sản phẩm') > -1 ||
    line.indexOf('Chọn sản phẩm') > -1
  ) {
    console.log(`Line ${i}: ${line.trim()}`);
  }
}
